lua/py-requirements/api.lua

```diff
@@ -63,4 +65,53 @@ function M.get_versions(name)
     return versions
 end
 
+---@class ModuleDescription
+---@field content string[]
+---@field content_type? string
+
+---@return ModuleDescription
+function M.parse_description(result)
```

Comment:
Let's combine some shared logic, I think the following helper function should do the trick:

```
---@param path string
---@param headers table<string,string>
---@return table|nil
local function call_pypi(path, headers)
    local result = curl.get(string.format('https://pypi.org/%s', path), {
        headers = vim.tbl_deep_extend('force', {
            ['User-Agent'] = 'py-requirements.nvim (https://github.com/MeanderingProgrammer/py-requirements.nvim)',
        }, headers),
        raw = { '--location' },
    })
    if result == nil or not vim.tbl_contains({ 200, 301 }, result.status) then
        return nil
    else
        return vim.json.decode(result.body)
    end
end
```

From there the functions can handle `nil` and use the body as needed. Can remove the parse functions and put those into the main api functions.